ink_linting/_Cargo.toml

```diff
@@ -28,11 +28,11 @@ regex = "1.5.4"
 dylint_testing = "2.0.0"
 
 # The following are ink! dependencies, they are only required for the `ui` tests.
-ink_primitives = { git = "https://github.com/paritytech/ink", default-features = false }
-ink_metadata = { git = "https://github.com/paritytech/ink", default-features = false, features = ["derive"] }
-ink_env = { git = "https://github.com/paritytech/ink",  default-features = false }
-ink_storage = { git = "https://github.com/paritytech/ink",  default-features = false }
-ink_lang = { git = "https://github.com/paritytech/ink", default-features = false }
+ink_primitives = { version = "3.3", default-features = false }
+ink_metadata = { version = "3.3", default-features = false, features = ["derive"] }
+ink_env = { version = "3.3",  default-features = false }
+ink_storage = { version = "3.3",  default-features = false }
+ink_lang = { version = "3.3", default-features = false }
```

Comment:
What is your thinking here? This way the CI for `cargo-contract` will no longer alert us if we merge something into ink! `master` that breaks the linter.

Reply:
This crate was introduced before the ink! 3.0 release, so pulling Git deps looks like a remnant of that. 

More importantly though, the lockfile is using old RC deps, which is what ends up getting used in the [`test-dylint` CI step](https://gitlab.parity.io/parity/mirrors/cargo-contract/-/jobs/1732092), not ink! `master`.

Reply:
I would leave it to reference the ink! `master`, otherwise we might publish an ink! release that in the worst case breaks the last `cargo-contract` release. We would only notice after having published the ink! release.

For the lockfile: Hmm, the CI is not using it. I guess we committed it to have `--locked` working and forgot to update it.

Reply:
It seems wrong to test it in that way. If you want to test for that it should be a CI job on the ink! repository.

Reply:
I'm with Alex here. We'd see the linter breaking in the `examples-contract-build` step of the ink! CI anyways